blog/2021/02/julia-1.6-highlights.md

```diff
@@ -0,0 +1,391 @@
+  4.995477 seconds …
+```
+Whereas the previous code loading precompilation process took ~8 minutes to
+precompile and load without indicating progress while doing so, the new
+mechanism took a little over 1 minute to precompile while showing progress through the
+dependencies, then the first time the package is loaded it happens at full
+speed. The new parallel precompilation process takes a depth-first approach of
+working through the dependency tree in the manifest, first precompiling the
+packages with no dependencies and working upward to the packages listed in the
+environment’s `Project.toml`, allowing multiple packages to be precompiled at
+the same time. The operation is multi-processed, as opposed to multi-threaded,
+so is not limited by Julia’s thread count. By default, Julia will spawn a
+maximum CPU-balanced load of package precompile jobs at once based on the number
+of CPU cores. Errors during precompilation will only throw for packages listed
+in the Project to allow for dependencies that may be listed in manifests but not
+loaded, and the auto precompilation process will remember if a package has
+errored within the given environment and will not retry until it changes.
+Auto-precompilation can be gracefully interrupted with a `ctrl-c` and disabled
+by setting the environment variable `JULIA_PKG_PRECOMPILE_AUTO=0`.
+
+~~~
+<script id="asciicast-381203" src="https://asciinema.org/a/381203.js" data-rows="25" data-cols="104" async></script>
+~~~
+
+For packages that are being developed, given that their code will be changed by
+other mechanisms than Pkg, this new workflow won’t automatically avoid
+encountering the standard code-load time precompilation. However non-dev-ed
+dependencies of those packages will be kept ready to load, so top-level
+precompilation at load time should remain lower for dev-ed packages.
+
+## Compile time percentage
+
+_Ian Butterworth_
+
+A small change that should help understanding of one of Julia’s quirks for
+newcomers is that the timing macro `@time` and its verbose friend `@timev` now
+report if any of the reported time has been spent on compilation.[^1]
+
+```julia-repl
+julia> x = rand(10,10);
+
+julia> @time x * x;
+  0.540600 seconds (2.35 M allocations: 126.526 MiB, 4.43% gc time, 99.94% compilation time)
+
+julia> @time x * x;
+  0.000010 seconds (1 allocation: 896 bytes)
+```
+Given Julia’s Just In Time (JIT) / Just Ahead Of Time (JAOT) compilation, the
+first time code is run the compilation overhead is often substantial, with big
+speed improvements seen in subsequent calls. This change highlights that
+behavior, serving as both a reminder and a tool for rooting out unwanted
+compilation effort i.e. over-specialized code.
+
+[^1]: Note that in some cases the system will look inside the `@time` expression
+and compile some of the called code before execution of the top-level expression begins.
+When that happens, some compilation time will not be counted.
```

Comment:
```suggestion
[^1]: Note that in some cases the system will look inside the `@time` expression and compile some of the called code before execution of the top-level expression begins. When that happens, some compilation time will not be counted. To include this time you can run `@time @eval ...`
```
Two changes: 
- Has to be a single line to be captured in the tip box
- Added tip to try `@time @eval ...` as per https://github.com/JuliaLang/julia/pull/39802